crates/server/src/authority/catalog.rs

```diff
@@ -429,33 +419,51 @@ async fn lookup<'a, R: ResponseHandler + Unpin>(
     request: &Request,
     response_edns: Option<Edns>,
     response_handle: R,
-) -> LookupControlFlow<ResponseInfo> {
+) -> Result<ResponseInfo, LookupError> {
+    let request_header = request.header();
+    let request_id = request.id();
+    let edns = request.edns();
+
     let query = request_info.query;
+
+    let lookup_options = lookup_options_for_edns(edns);
+
+    // log algorithms being requested
+    if lookup_options.dnssec_ok() {
+        info!(
+            "request: {} lookup_options: {:?}",
+            request_id, lookup_options
+        );
+    }
+
     debug!(
         "request: {} found authority: {}",
         request.id(),
         authority.origin()
     );
 
-    let response = build_response(
-        authority,
-        request_info,
-        request.id(),
-        request.header(),
-        query,
-        request.edns(),
-    )
-    .await;
+    debug!("performing {} on {}", query, authority.origin());
 
-    let is_continue = response.is_continue();
+    // Wait so we can determine if we need to fire a request to the next authority in a chained
+    // configuration if the current authority declines to answer. NOTE: This is in preparation
+    // for the chained authority PR.
+    let result = authority.search(request_info, lookup_options).await;
 
-    let (response_header, sections) = match response {
-        Continue(Ok(lookup)) | Break(Ok(lookup)) => lookup,
-        Continue(Err(e)) => return Continue(Err(e)),
-        Break(Err(e)) => return Break(Err(e)),
-        Skip => return Skip,
+    // Abort only if the authority declined to handle the request.
+    if let LookupControlFlow::Skip = result {
+        trace!("build_response: aborting search on lookupcontrolflow::skip");
```

Comment:
Sorry, I didn't mean that log messages should be all-lowercase, just that we shouldn't capitalize them. I think we should spell `LookupControlFlow::Skip` "correctly" (as should other names).